scrypto-unit/src/test_runner.rs

```diff
@@ -235,24 +235,38 @@ impl CustomGenesis {
     }
 }
 
-pub struct TestRunnerBuilder {
+pub trait Database:
+    SubstateDatabase + CommittableSubstateDatabase + ListableSubstateDatabase
+{
+}
+impl<T: SubstateDatabase + CommittableSubstateDatabase + ListableSubstateDatabase> Database for T {}
+
+pub type DefaultTestRunner = TestRunner<NoExtension, InMemorySubstateDatabase>;
+
+pub struct TestRunnerBuilder<E, D> {
     custom_genesis: Option<CustomGenesis>,
+    custom_extension: E,
+    custom_database: D,
     trace: bool,
     state_hashing: bool,
 }
 
-impl TestRunnerBuilder {
-    pub fn new() -> TestRunnerBuilder {
+impl TestRunnerBuilder<NoExtension, InMemorySubstateDatabase> {
```

Comment:
Cleaner may be `impl DefaultTestRunner {`

Reply:
I had the same thought!
...but this is a `TestRunnerBuilder`, not `TestRunner` :(

Reply:
ah!